### ISSUE TITLE
Default Unit Tests do not pass

### ISSUE THREAD

**Issue by Perseverance** [smartcontractkit/chainlink#1403]
Hey guys, I am running the tests on a plain newly unboxed box and e of the default tests are not passing:

```
AssertionError: expected following error message to include "invalid JUMP" or "revert": "Invalid number of parameters for "fulfill". Got 3 expected 6!"
...
AssertionError: expected following error message to include "invalid JUMP" or "revert": "Invalid number of parameters for "withdrawLink". Got 1 expected 2!"
...
Error: Invalid number of parameters for "withdrawLink". Got 1 expected 2!
```

Any idea what goes wrong?

**Comment by thodges-gh** [comment]
@Perseverance What OS are you using (so that I can try to reproduce this)? And just making sure, this is on the develop branch?

**Comment by Perseverance** [comment]
I will close this issue and move it to the truffle box I used. Sorry for the confusion caused here

**Comment by Perseverance** [comment]
Actually I noticed I cannot open an issue here thus I will this one open. It is regarding the truffle box. I used the following command.

`truffle unbox smartcontractkit/chainlink`


My OS is OSX 10.14.5

**Comment by thodges-gh** [comment]
@Perseverance What version of Truffle are you using?

**Comment by Perseverance** [comment]
`Truffle v5.0.18`. Just updated to `Truffle v5.0.24`, but no luck here either. For what it is worth I am using `etherlime`s ganache - `etherlime ganache`. It is just a regular ganache under the hood.

**Comment by thodges-gh** [comment]
Sorry for the delay. Can you provide more info on how you're running the tests? For example, below I update Truffle to the latest (currently 5.0.25) and run `npm test` without issue. If you're making changes to the `truffle-config.js` file, please include that here. My assumption is that your issue has to do with `etherlime ganache`.

```
 thomas@thomas-Precision-5510  ~/truffletest1  npm install -g truffle
/home/thomas/.nvm/versions/node/v10.15.1/bin/truffle -> /home/thomas/.nvm/versions/node/v10.15.1/lib/node_modules/truffle/build/cli.bundled.js

> truffle@5.0.25 postinstall /home/thomas/.nvm/versions/node/v10.15.1/lib/node_modules/truffle
> node ./scripts/postinstall.js

- Fetching solc version list from solc-bin. Attempt #1
+ truffle@5.0.25
removed 64 packages and updated 9 packages in 9.065s


   ╭───────────────────────────────────────────────────────────────╮
   │                                                               │
   │       New patch version of npm available! 6.9.0 → 6.9.2       │
   │   Changelog: https://github.com/npm/cli/releases/tag/v6.9.2   │
   │               Run npm install -g npm to update!               │
   │                                                               │
   ╰───────────────────────────────────────────────────────────────╯

 thomas@thomas-Precision-5510  ~/truffletest1  truffle unbox smartcontractkit/chainlink

✔ Preparing to download
✔ Downloading
✔ Cleaning up temporary files
✔ Setting up box

Unbox successful. Sweet!

Commands:

  Run tests:                      npm test
  Compile contracts:              npm run compile
  Migrate to development network: npm run migrate:dev
  Migrate to live network:        npm run migrate:live

 thomas@thomas-Precision-5510  ~/truffletest1  npm test

> chainlink-box@0.6.0 test /home/thomas/truffletest1
> truffle test


Compiling your contracts...
===========================
> Compiling ./contracts/Migrations.sol
> Compiling ./contracts/MyContract.sol
> Compiling ./contracts/lib/LinkToken.sol
> Compiling ./contracts/lib/Oracle.sol
> Compiling @ensdomains/buffer/contracts/Buffer.sol
> Compiling chainlink/contracts/Chainlink.sol
> Compiling chainlink/contracts/ChainlinkClient.sol
> Compiling chainlink/contracts/ENSResolver.sol
> Compiling chainlink/contracts/Oracle.sol
> Compiling chainlink/contracts/interfaces/ChainlinkRequestInterface.sol
> Compiling chainlink/contracts/interfaces/ENSInterface.sol
> Compiling chainlink/contracts/interfaces/LinkTokenInterface.sol
> Compiling chainlink/contracts/interfaces/OracleInterface.sol
> Compiling chainlink/contracts/interfaces/PointerInterface.sol
> Compiling link_token/contracts/ERC677Token.sol
> Compiling link_token/contracts/LinkToken.sol
> Compiling link_token/contracts/math/linkSafeMath.sol
> Compiling link_token/contracts/token/ERC677.sol
> Compiling link_token/contracts/token/ERC677Receiver.sol
> Compiling link_token/contracts/token/linkBasicToken.sol
> Compiling link_token/contracts/token/linkERC20.sol
> Compiling link_token/contracts/token/linkERC20Basic.sol
> Compiling link_token/contracts/token/linkStandardToken.sol
> Compiling openzeppelin-solidity/contracts/math/SafeMath.sol
> Compiling openzeppelin-solidity/contracts/ownership/Ownable.sol
> Compiling solidity-cborutils/contracts/CBOR.sol

    > compilation warnings encountered:

link_token/contracts/token/linkStandardToken.sol:27:5: Warning: Use of the "var" keyword is deprecated.
    var _allowance = allowed[_from][msg.sender];
    ^------------^
,link_token/contracts/LinkToken.sol:15:3: Warning: Defining constructors as functions with the same name as the contract is deprecated. Use "constructor(...) { ... }" instead.
  function LinkToken()
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/ERC677Token.sol:21:5: Warning: Invoking events without "emit" prefix is deprecated.
    Transfer(msg.sender, _to, _value, _data);
    ^--------------------------------------^
,link_token/contracts/token/linkBasicToken.sol:25:5: Warning: Invoking events without "emit" prefix is deprecated.
    Transfer(msg.sender, _to, _value);
    ^-------------------------------^
,link_token/contracts/token/linkStandardToken.sol:35:5: Warning: Invoking events without "emit" prefix is deprecated.
    Transfer(_from, _to, _value);
    ^--------------------------^
,link_token/contracts/token/linkStandardToken.sol:46:5: Warning: Invoking events without "emit" prefix is deprecated.
    Approval(msg.sender, _spender, _value);
    ^------------------------------------^
,link_token/contracts/token/linkStandardToken.sol:69:5: Warning: Invoking events without "emit" prefix is deprecated.
    Approval(msg.sender, _spender, allowed[msg.sender][_spender]);
    ^-----------------------------------------------------------^
,link_token/contracts/token/linkStandardToken.sol:81:5: Warning: Invoking events without "emit" prefix is deprecated.
    Approval(msg.sender, _spender, allowed[msg.sender][_spender]);
    ^-----------------------------------------------------------^
,link_token/contracts/token/linkERC20Basic.sol:11:3: Warning: No visibility specified. Defaulting to "public".
  function balanceOf(address who) constant returns (uint256);
  ^---------------------------------------------------------^
,link_token/contracts/token/linkERC20Basic.sol:12:3: Warning: No visibility specified. Defaulting to "public".
  function transfer(address to, uint256 value) returns (bool);
  ^----------------------------------------------------------^
,link_token/contracts/token/linkERC20.sol:12:3: Warning: No visibility specified. Defaulting to "public".
  function allowance(address owner, address spender) constant returns (uint256);
  ^----------------------------------------------------------------------------^
,link_token/contracts/token/linkERC20.sol:13:3: Warning: No visibility specified. Defaulting to "public".
  function transferFrom(address from, address to, uint256 value) returns (bool);
  ^----------------------------------------------------------------------------^
,link_token/contracts/token/linkERC20.sol:14:3: Warning: No visibility specified. Defaulting to "public".
  function approve(address spender, uint256 value) returns (bool);
  ^--------------------------------------------------------------^
,link_token/contracts/token/ERC677.sol:6:3: Warning: No visibility specified. Defaulting to "public".
  function transferAndCall(address to, uint value, bytes data) returns (bool success);
  ^----------------------------------------------------------------------------------^
,link_token/contracts/token/ERC677Receiver.sol:5:3: Warning: No visibility specified. Defaulting to "public".
  function onTokenTransfer(address _sender, uint _value, bytes _data);
  ^------------------------------------------------------------------^
,link_token/contracts/token/linkBasicToken.sol:22:3: Warning: No visibility specified. Defaulting to "public".
  function transfer(address _to, uint256 _value) returns (bool) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/token/linkBasicToken.sol:34:3: Warning: No visibility specified. Defaulting to "public".
  function balanceOf(address _owner) constant returns (uint256 balance) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/token/linkStandardToken.sol:26:3: Warning: No visibility specified. Defaulting to "public".
  function transferFrom(address _from, address _to, uint256 _value) returns (bool) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/token/linkStandardToken.sol:44:3: Warning: No visibility specified. Defaulting to "public".
  function approve(address _spender, uint256 _value) returns (bool) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/token/linkStandardToken.sol:56:3: Warning: No visibility specified. Defaulting to "public".
  function allowance(address _owner, address _spender) constant returns (uint256 remaining) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/token/linkStandardToken.sol:66:3: Warning: No visibility specified. Defaulting to "public".
  function increaseApproval (address _spender, uint _addedValue)
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/token/linkStandardToken.sol:73:3: Warning: No visibility specified. Defaulting to "public".
  function decreaseApproval (address _spender, uint _subtractedValue)
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/ERC677Token.sol:38:3: Warning: Function state mutability can be restricted to view
  function isContract(address _addr)
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/math/linkSafeMath.sol:9:3: Warning: Function state mutability can be restricted to pure
  function mul(uint256 a, uint256 b) internal constant returns (uint256) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/math/linkSafeMath.sol:15:3: Warning: Function state mutability can be restricted to pure
  function div(uint256 a, uint256 b) internal constant returns (uint256) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/math/linkSafeMath.sol:22:3: Warning: Function state mutability can be restricted to pure
  function sub(uint256 a, uint256 b) internal constant returns (uint256) {
  ^ (Relevant source part starts here and spans across multiple lines).
,link_token/contracts/math/linkSafeMath.sol:27:3: Warning: Function state mutability can be restricted to pure
  function add(uint256 a, uint256 b) internal constant returns (uint256) {
  ^ (Relevant source part starts here and spans across multiple lines).

> Artifacts written to /tmp/test-119529-31510-ef80no.z0pi
> Compiled successfully using:
   - solc: 0.4.24+commit.e67f0147.Emscripten.clang



  Contract: MyContract
    #createRequest
      without LINK
        ✓ reverts (147ms)
      with LINK
        sending a request to a specific oracle contract address
          ✓ triggers a log event in the new Oracle contract (93ms)
    #fulfill
      ✓ records the data given to it by the oracle
      when my contract does not recognize the request ID
        ✓ does not accept the data provided
      when called by anyone other than the oracle contract
        ✓ does not accept the data provided
    #cancelRequest
      before the expiration time
        ✓ cannot cancel a request (42ms)
      after the expiration time
        when called by a non-owner
          ✓ cannot cancel a request
        when called by an owner
          ✓ can cancel a request (38ms)
    #withdrawLink
      when called by a non-owner
        ✓ cannot withdraw
      when called by the owner
        ✓ transfers LINK to the owner (61ms)


  10 passing (3s)

 thomas@thomas-Precision-5510  ~/truffletest1  truffle version
Truffle v5.0.25 (core: 5.0.25)
Solidity - 0.4.24 (solc-js)
Node v10.15.1
Web3.js v1.0.0-beta.37
```

**Comment by Perseverance** [comment]
Hey @thodges-gh , my colleague has tested this out and it seems that it works fine on her machine so I will close this one. Expect maybe some more from us as we are building etherlime shape for chainlink ;)
